analysis/coverage_data_utils.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utility functions for coverage data calculation."""
+
+import posixpath
+from collections import defaultdict
+import tempfile
+import json
+import os
+import pandas as pd
+
+from common import filestore_utils
+
+
+def get_fuzzer_benchmark_key(fuzzer: str, benchmark: str):
+    """Returns the key in coverage dict for a pair of fuzzer-benchmark."""
+    return fuzzer + ' ' + benchmark
+
+
+def get_fuzzer_filestore_path(benchmark_df, fuzzer):
+    """Gets the filestore_path for |fuzzer| in |benchmark_df|."""
+    fuzzer_df = benchmark_df[benchmark_df.fuzzer == fuzzer]
```

Comment:
This is supposed to work with merging experiments right?
Can you confirm that it does? Looking at this function, it feels like intuitively it does not.

Reply:
Yes I'll test it!

Reply:
Like I guess this sort of assumes that benchmarks for a fuzzer all come from the same experiment. I don't think we make that assumption elsewhere though. I think the assumption is fine to make though it might break in some rare cases (such as when we add benchmarks). Can you at least document this assumption please?

Reply:
I just tested and it works. The `benchmark_df` here is the dataframe for one certain benchmark. So we are not assuming all benchmarks for a fuzzer come from the same experiment. We are just assuming all trials for a certain pair of fuzzer-benchmark come from the same experiment, which is what `merge_with_clobber` does.